### ISSUE TITLE
Fix tests for XHR

### ISSUE THREAD

**Issue by dvoytenko** [ampproject/amphtml#5040]
```
Chromium 51.0.2704 (Ubuntu 0.0.0) XHR Polyfill should NOT succeed CORS setting cookies without credentials FAILED
	AssertionError: expected 'v1' to be null
	    at doAsserterAsyncAndAddThen (/home/travis/build/ampproject/amphtml/node_modules/chai-as-promised/lib/chai-as-promised.js:292:33)
	    at .<anonymous> (/home/travis/build/ampproject/amphtml/node_modules/chai-as-promised/lib/chai-as-promised.js:282:25)
	    at get (/home/travis/build/ampproject/amphtml/node_modules/chai/chai.js:5396:37)
	    at /tmp/f7c58e12910b6902c4d675cb4b5c39ac.browserify:120064:71 <- /home/travis/build/ampproject/amphtml/test/functional/test-xhr.js:367:53
```

**Comment by mkhatib** [comment]
This actually might be side effect of the wrong stubbing of `Math.random` since we set that cookie in a different test.

**Comment by mkhatib** [comment]
Fixed in #5042
